shaders/Membrane.shader

```diff
@@ -25,16 +29,26 @@ void vertex(){
 }
 
 void fragment(){
-    vec4 normal = texture(albedoTexture, UV);
+    vec4 albedo = texture(albedoTexture, UV);
+    
+    vec4 normalmap = texture(normalTexture, UV);
+    vec3 normals = vec3 (normalmap.r * 2.0 - 1.0, normalmap.g * 2.0 - 1.0, 
+    normalmap.b * 2.0 - 1.0);
+
+    float y = normals.y;
+    normals.y = -normals.z;
+    normals.z = y;
+
     vec4 damaged = texture(damagedTexture, UV);
-    vec4 final = ((normal * healthFraction) + 
-        (damaged * (1.f - healthFraction))) * tint;
+    vec4 final = vec4 ((((albedo.rgb * healthFraction) + 
+        (damaged.rbg * (1.f - healthFraction))) * tint.rgb), albedo.a);
```

Comment:
Why is the damaged texture alpha now ignored? (If I understood this change correctly...)

Reply:
If I am remembering right, it was unnecessary and just messed with shading,(or I was messing with stuff and forgot to change it back) i'll have to look again though.

Reply:
My concern is that if the crack parts in the damaged texture are at locations where the undamaged texture is very transparent, the damage won't show up well. And this limitation of needing the same alpha in the damaged as the undamaged texture, probably will limit art choices etc.

Reply:
> My concern is that if the crack parts in the damaged texture are at locations where the undamaged texture is very transparent, the damage won't show up well. And this limitation of needing the same alpha in the damaged as the undamaged texture, probably will limit art choices etc.

now I remember after looking, the person before combined Alpha values very incorrectly, you're supposed to use mix(x, y, 1.0) not just add them together. I added this just now on my local branch.

Reply:
Don't you need to take the healthFraction into account in the mixing, though?
Wasn't this math right, since it took the healtFraction of alpha from the undamaged texture and (1 - healthFraction) from the damaged one? That results in alpha being 1 if healthFraction is 1 and the undamaged texture is opaque, and otherwise a mix between the two alpha values?

Reply:
> Don't you need to take the healthFraction into account in the mixing, though?
> Wasn't this math right, since it took the healtFraction of alpha from the undamaged texture and (1 - healthFraction) from the damaged one? That results in alpha being 1 if healthFraction is 1 and the undamaged texture is opaque, and otherwise a mix between the two alpha values?

it does with the RGB but actually now that I rethink and reread, it seems you are correct, the old math is fine, i guess there was something else i'm just not remembering. Anyways it seems the old stuff was fine so i'll revert.